common/game-state.hpp

```diff
@@ -21,15 +21,32 @@ namespace tec {
 
 		GameState() : state_id(0) { }
 
-		GameState(const GameState&) = delete;
+		GameState(const GameState& other) {
+			if (&other == this) {
+				return;
+			}
+			this->positions = other.positions;
+			this->orientations = other.orientations;
+			this->velocities = other.velocities;
+			this->state_id = other.state_id;
+		}
 		GameState(GameState&& other) {
 			this->positions = std::move(other.positions);
 			this->orientations = std::move(other.orientations);
 			this->velocities = std::move(other.velocities);
 			this->state_id = other.state_id;
 		}
 
-		GameState& operator=(const GameState& other) = delete;
+		GameState& operator=(const GameState& other) {
```

Comment:
Please check for the future whether this can just be `GameState& operator=(const GameState& other) = default`, that'd make it a bit nicer. Same with the other ones.